### PR TITLE
Enable building and running kubevirt on s390x

### PR DESCRIPTION
**Pull request by vamsikrishna-siddu**
### What this PR does
Before this PR:
  There is no support of s390x architecture in kubevirt

After this PR:
  kubevirt is supported on s390x architecture.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Special notes for your reviewer
   We are not able to rebase the existing PR https://github.com/kubevirt/kubevirt/pull/10490 raised by @jschintag  which adds the support of s390x in kubevirt due to the unavailability of the contributor. So I have raised this PR with the same changes of the mentioned PR by rebasing with the latest changes and and addressing this comment.  https://github.com/kubevirt/kubevirt/pull/10490#issuecomment-2149365438

 
**TODOS:**
- [x]  Test all changed build scripts on s390x
- [x]  Rework Graphic device
- [x]  Disable ACPI by default on s390x
- [x]  Add s390x to tests
- [x]  Enable cross-compiling
- [x]  Do not build every container image on s390x, as some base images are not available
- [x]  Cleanup commits (Squash, wrong author etc.)
- [x]  Rebase PR on main
- [x]  (Optional) Split PR into multiple smaller PRs
- [x]  Write Release Note

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [x] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
 Add support for building and running kubevirt on s390x.
```

### CODE REVIEW

**Comment by alicefr** [comment]
Looks good
/approve
@andreabolognani can you please take another look?

**Review by andreabolognani — COMMENTED** [review]
@vamsikrishna-siddu I would prefer it if the original PR was used. There should be just enough time to make it happen before feature freeze.

At the very least, the issues I've mentioned need to be addressed. See https://github.com/andreabolognani/kubevirt/commits/enable-s390x/ for a branch where that happens.

Notice how I've ensured that @jschintag would retain sole ownership of all the commits by running

```
$ git rebase main -x 'EDITOR=true GIT_COMMITTER_NAME="Jan Schintag" GIT_COMMITTER_EMAIL="jan.schintag@de.ibm.com" git commit --amend'
```

before pushing the branch.

File: pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator_test.go [hunk 1]
```diff
@@ -185,7 +185,7 @@ var _ = Describe("VirtualMachine Mutator", func() {
 					ArchitectureConfiguration: &v1.ArchConfiguration{
 						Amd64:   &v1.ArchSpecificConfiguration{MachineType: amd64MachineType},
 						Arm64:   &v1.ArchSpecificConfiguration{MachineType: arm64MachineType},
-						Ppc64le: &v1.ArchSpecificConfiguration{MachineType: ppc64leMachineType},
```
[the remark this answers is not in the document]
Comment (vamsikrishna-siddu) [reply pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator_test.go]: done

File: pkg/virt-launcher/virtwrap/converter/converter.go [hunk 2]
```diff
@@ -2085,7 +2085,7 @@ func GracePeriodSeconds(vmi *v1.VirtualMachineInstance) int64 {
 }
 
 func InterpretTransitionalModelType(useVirtioTransitional *bool, archString string) string {
-	// "virtio-non-transitional" and "virtio-transitional" are both PCI devices, so they don't work on s390x.
```
[the remark this answers is not in the document]
Comment (vamsikrishna-siddu) [reply pkg/virt-launcher/virtwrap/converter/converter.go]: done

**Comment by vamsikrishna-siddu** [comment]
Hi @andreabolognani , I have addressed your review comments. Can you please review it again thanks.

**Comment by andreabolognani** [comment]
@vamsikrishna-siddu looks good now, thanks.

I still think it would be better if the original PR (#10490) was merged instead of this one, primarily so that Jan is properly credited in the release notes for all the time and effort he's been putting into s390x support over the past several months.

A way to do that without introducing further delays and increase our chances of this work making it into `main` before feature freeze would be for a maintainer to force-push to @jschintag's branch.

@xpivarc @vladikr what do you think? Would you be okay with updating the original PR using your maintainer super powers, or would you rather merge this PR instead?

**Comment by vamsikrishna-siddu** [comment]
Hello everyone as the original PR https://github.com/kubevirt/kubevirt/pull/10490 was rebased and merged into the main. This PR is no longer needed. Hence Iam closing this PR. Thanks

**Pull request closed without merging**